.github/workflows/dev_build_test.yml

```
@@ -0,0 +1,61 @@
+name: HTTomo dev build
+
+on:
+  push:
+    branches:
+      - dev
+  pull_request:
+    branches:
+      - dev
+  
+jobs:
+
+  install-httomo-dev-test-iris:
+    runs-on: iris-gpu
+    container:
+      image: nvidia/cuda:11.6.2-devel-ubi8
+      env:
+        NVIDIA_VISIBLE_DEVICES: ${{ env.NVIDIA_VISIBLE_DEVICES }}
+
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
```

```suggestion
        uses: actions/checkout@v4
```
Not so important but, you can fix the warning [here](https://github.com/DiamondLightSource/httomo/actions/runs/9894014080) with this change.

---

cheers, will fix that in other files as well.